geom/alg_point_in_ring.go

```diff
@@ -42,25 +46,63 @@ func pointRingSide(pt XY, ring LineString) side {
 		if !ok {
 			continue
 		}
-		inter := ln.intersectLine(ray)
-		if inter.empty {
-			continue
+		if incrementCountPointInRing(pt, ray, ln) {
+			count++
 		}
-		if inter.ptA != inter.ptB {
-			continue
+	}
+	if count%2 == 1 {
+		return interior
+	}
+	return exterior
+}
+
+func incrementCountPointInRing(pt XY, ray, iterLine line) bool {
+	inter := ray.intersectLine(iterLine)
+	if inter.empty {
+		return false
+	}
+	if inter.ptA != inter.ptB {
+		return false
+	}
+	if inter.ptA == iterLine.a || inter.ptA == iterLine.b {
+		otherY := iterLine.a.Y
+		if inter.ptA == iterLine.a {
+			otherY = iterLine.b.Y
 		}
-		if inter.ptA == ln.a || inter.ptA == ln.b {
-			otherY := ln.a.Y
-			if inter.ptA == ln.a {
-				otherY = ln.b.Y
-			}
-			if otherY < pt.Y {
-				count++
-			}
-		} else {
-			count++
+		return otherY < pt.Y
+	}
+	return true
+}
+
+func relatePointToPolygon(pt XY, polyBoundary indexedLines) side {
```

Comment:
Need to add some additional tests cases around this (could potentially modify the existing test suite for the non-indexed version of this algorithm).

Reply:
Additional tests added in 55c7cd9